molecules/README.md

```diff
@@ -0,0 +1,169 @@
+# Molecules
+This sample shows how to train and create an ML model to predict the energy on molecules, including how to preprocess raw data files.
+
+The dataset for this sample comes from this [Kaggle Dataset](https://www.kaggle.com/burakhmmtgl/predict-molecular-properties). However, the number of preprocessed JSON files was too small, so this sample will download the raw data files directly from this [FTP source](ftp.ncbi.nlm.nih.gov/pubchem/Compound_3D/01_conf_per_cmpd/SDF) (in `sdf` fromat instead of `json`)
+
+Roughly the steps are the following:
+ 1. Validate the command line arguments
+ 2. Extract the raw data
+ 3. Preprocessing ([Apache Beam](https://beam.apache.org/) and [tf.Transform](https://github.com/tensorflow/transform)), this can be run distributed in [Google Cloud Dataflow](https://cloud.google.com/dataflow/) 
+ 4. Training ([Tensorflow](https://www.tensorflow.org/)), this can be run distributed in [Google Cloud ML Engine](https://cloud.google.com/ml-engine/)
+
+## Installing
+This requires `python2`, Apache Beam does not currently support `python3`.
+
+### Getting the source code
+You can clone the github repository and then navigate to the `molecules` sample directory. The rest of the instructions assume that you are in that directory.
+```bash
+git clone https://github.com/GoogleCloudPlatform/cloudml-samples.git
+cd cloudml-samples/molecules
+```
+
+### Python virtual environment
+Using [virtualenv](https://virtualenv.pypa.io/en/stable/) to isolate your dependencies is recommended. To set up make
+sure you have the `virtualenv` package installed.
+```bash
+pip install --user virtualenv
+```
+
+To create and activate a new virtual environment, run the following commands:
+```bash
+python -m virtualenv env
+source env/bin/activate
+```
+
+To deactivate the virtual environment, run:
+```bash
+deactivate
+```
+
+If you do not want to use `virtualenv`, you can use the `--user` flag when
```

Comment:
We should make exclusive virtualenv recommendation here and remove --user flag option. See https://packaging.python.org/guides/installing-using-pip-and-virtualenv/#installing-virtualenv for details.